### DISCUSSION TITLE
Upcoming deprecations and supported platform changes

### DISCUSSION THREAD

**Discussion by charlie-foxtrot** [rtl-airband/RTLSDR-Airband#447] [General]
I'm planning to deprecate explicit build support for `rpiv1` (Raspberry Pi version 1 - ARMv6 CPU, Broadcom VideoCore IV GPU) in an upcoming release as I have no hardware to test it.

Is anyone using `rtl_airband` on an original raspberry pi?

I plan to continue to support (and test) `rpiv2` (Raspberry Pi version 2 or 3 - ARMv7 CPU, Broadcom VideoCore IV GPU) _**assuming**_ it is currently working with the latest Raspberry Pi OS (bookworm).

I am considering further reducing the set of build platforms by replacing `armv7-generic` and `armv8-generic` with a single `generic` option that works across x86, arm, etc.

Does anyone use `armv7-generic` or `armv8-generic` today?

Finally, under the assumption that the vast majority of people are compiling on the device they are running on, I plan to change the default platform to `native`.

**Comment by charlie-foxtrot** [comment]
these changes have been made in #444 and will be part of `v5.0.0`

**Reply by tjharman** [reply]
I missed this - I am running on a Raspberry Pi Zero W Rev 1.1 which is ARMv6

```
processor       : 0
model name      : ARMv6-compatible processor rev 7 (v6l)
BogoMIPS        : 997.08
Features        : half thumb fastmult vfp edsp java tls
CPU implementer : 0x41
CPU architecture: 7
CPU variant     : 0x0
CPU part        : 0xb76
CPU revision    : 7

Hardware        : BCM2835
Revision        : 9000c1
Serial          : 0000000019ce26ec
Model           : Raspberry Pi Zero W Rev 1.1
```

Still, I understand not supporting hardware you don't have/can't test.  The current version is working well for me.  The Broadcom offload means it only uses ~50% CPU which I think is amazing given what it's doing.

So in summary yes I have a old CPU but I fully understand and agree with your stance to drop support for it.  Thank you for a great project.

**Reply by charlie-foxtrot** [reply]
No problem, the new default platform should work fine. 

It's only the use of the GPU for FFT on the rpi1s that's no longer supported. 

Can you try updating to the latest main and do a clean build?

**Reply by charlie-foxtrot** [reply]
If the performance is much worse, let me know. If there's a good argument to continue supporting (and I can test it) then I will
